lms/views/lti/deeplinking.py

```diff
@@ -0,0 +1,91 @@
+import json
+from urllib.parse import urlencode, urlparse
+
+from pyramid.view import view_config, view_defaults
+from webargs import fields
+
+from lms.services import LTIAHTTPService
+from lms.validation._base import JSONPyramidRequestSchema
+
+
+class DeeplinkingSignRequestSchema(JSONPyramidRequestSchema):
+    deep_linking_settings = fields.Dict(required=False)
+    content_item_return_url = fields.Str(required=True)
+    content = fields.Dict(required=True)
+
+    extra_params = fields.Dict(required=False)
+
+
+@view_defaults(
+    request_method="POST", renderer="json", schema=DeeplinkingSignRequestSchema
+)
+class DeeplinkingParametersViews:
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    @view_config(route_name="lti.deeplinking.formfields")
+    def deeplinking_payload_v13(self):
+        ltia_service = self.request.find_service(LTIAHTTPService)
+        url = self._content_to_url(
+            self.request.parsed_params["content"],
+            self.request.parsed_params["extra_params"],
+        )
+
+        ltia_signature = ltia_service.sign(
+            {
+                "https://purl.imsglobal.org/spec/lti/claim/deployment_id": self.request.find_service(
+                    name="application_instance"
+                )
+                .get_current()
+                .deployment_id,
+                "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiDeepLinkingResponse",
+                "https://purl.imsglobal.org/spec/lti/claim/version": "1.3.0",
+                "https://purl.imsglobal.org/spec/lti-dl/claim/content_items": [
+                    {
+                        "type": "ltiResourceLink",
+                        "url": url,
+                    }
+                ],
+                "https://purl.imsglobal.org/spec/lti-dl/claim/data": self.request.parsed_params[
+                    "deep_linking_settings"
+                ],
+            }
+        )
+        return {"JWT": ltia_signature}
+
+    @view_config(route_name="lti.v11.deeplinking.formfields")
+    def deeplinking_payload_v11(self):
+        url = self._content_to_url(
+            self.request.parsed_params["content"],
+            self.request.parsed_params["extra_params"],
+        )
+
+        return {
+            "content_items": json.dumps(
+                {
+                    "@context": "http://purl.imsglobal.org/ctx/lti/v1/ContentItem",
+                    "@graph": [
+                        {
+                            "@type": "LtiLinkItem",
+                            "mediaType": "application/vnd.ims.lti.v1.ltilink",
+                            "url": url,
+                        },
+                    ],
+                }
+            )
+        }
+
+    def _content_to_url(self, content, extra_params):
```

Comment:
For deeplinking we might produce an URL back to our LTI launch endpoint including extra info about the document.